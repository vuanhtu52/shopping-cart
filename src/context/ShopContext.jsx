import { createContext } from "react";

const ShopContext = createContext({
    products: [],
    cartItems: [],
    addToCart: () => {},
    addMultipleToCart: () => {},
    removeOneFromCart: () => {},
    getTotalCartItems: () => {},
});

export default ShopContext;