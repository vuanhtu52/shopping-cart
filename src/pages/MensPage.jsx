const MensPage = () => {
    return <div>Men's clothing page</div>;
};

export default MensPage;
