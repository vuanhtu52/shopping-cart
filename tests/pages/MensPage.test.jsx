import { render, screen } from "@testing-library/react";
import ShopContext from "../../src/context/ShopContext";
import { MemoryRouter } from "react-router-dom";
import { describe } from "vitest";
import MensPage from "../../src/pages/MensPage";

const renderComponent = products => {
    return render(
        <ShopContext.Provider value={{products}}>
            <MemoryRouter>
                <MensPage />
            </MemoryRouter>
        </ShopContext.Provider>
    );
};

describe("MensPage component", () => {
    it("Renders heading correctly", () => {
        renderComponent([]);
        expect(screen.getByText("Men's Clothing")).toBeInTheDocument();
    });

    it("Renders only men's clothing", () => {
        const products = [
            { id: 1, title: "Shirt", category: "men's clothing", price: 20, image: "shirt.jpg" },
            { id: 2, title: "Pants", category: "men's clothing", price: 40, image: "pants.jpg" },
            { id: 3, title: "Dress", category: "women's clothing", price: 30, image: "dress.jpg" },
        ];

        renderComponent(products);

        expect(screen.getByText("Shirt")).toBeInTheDocument();
        expect(screen.getByText("Pants")).toBeInTheDocument();
        expect(screen.queryByText("Dress")).not.toBeInTheDocument();
    });

    it("Renders no products when none are jewellery", () => {
        const products = [
            { id: 1, title: "Dress", category: "women's clothing", price: 30, image: "dress.jpg" },
            { id: 2, title: "Skirt", category: "women's clothing", price: 25, image: "skirt.jpg" },
        ];

        renderComponent(products);

        expect(screen.queryByText("Dress")).not.toBeInTheDocument();
        expect(screen.queryByText("Skirt")).not.toBeInTheDocument();
    });

    it("Snapshot testing", () => {
        const products = [
            { id: 1, title: "Shirt", category: "men's clothing", price: 20, image: "shirt.jpg" },
            { id: 2, title: "Pants", category: "men's clothing", price: 40, image: "pants.jpg" },
        ];

        const { container } = renderComponent(products);
        expect(container).toMatchSnapshot();
    });
});