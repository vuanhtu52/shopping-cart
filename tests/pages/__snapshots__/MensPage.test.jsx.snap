// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MensPage component > Snapshot testing 1`] = `
<div>
  <div>
    <h2
      class="text-center text-xl font-bold p-4"
    >
      Men's Clothing
    </h2>
    <div
      class="grid gap-8 justify-items-center grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 p-4"
    >
      <div
        class="flex flex-col min-w-60 w-60 gap-2 border border-gray-300 p-2 rounded-2xl"
      >
        <a
          href="/product/1"
        >
          <img
            alt="Shirt"
            class="object-contain w-full min-h-64 h-64 p-4"
            src="shirt.jpg"
          />
        </a>
        <a
          class="hover:text-[#fb923c]"
          href="/product/1"
        >
          Shirt
        </a>
        <div
          class="font-bold text-lg"
        >
          $
          20
        </div>
        <button
          class="self-start bg-amber-300 p-2 rounded-xl text-sm"
        >
          Add to cart
        </button>
      </div>
      <div
        class="flex flex-col min-w-60 w-60 gap-2 border border-gray-300 p-2 rounded-2xl"
      >
        <a
          href="/product/2"
        >
          <img
            alt="Pants"
            class="object-contain w-full min-h-64 h-64 p-4"
            src="pants.jpg"
          />
        </a>
        <a
          class="hover:text-[#fb923c]"
          href="/product/2"
        >
          Pants
        </a>
        <div
          class="font-bold text-lg"
        >
          $
          40
        </div>
        <button
          class="self-start bg-amber-300 p-2 rounded-xl text-sm"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</div>
`;
